pkg/argsbuilder/argsbuilder_args.go

```diff
@@ -15,13 +18,59 @@ type Value = string
 // Args represents a set of args.
 type Args map[Key]Value
 
+// MustMerge implements the ArgsBuilder interface.
+func (a Args) MustMerge(args Args, setters ...MergeOption) {
+	if err := a.Merge(args, setters...); err != nil {
+		panic(err)
+	}
+}
+
 // Merge implements the ArgsBuilder interface.
-func (a Args) Merge(args Args) ArgsBuilder {
+//nolint:gocyclo
+func (a Args) Merge(args Args, setters ...MergeOption) error {
+	var opts MergeOptions
+
+	for _, s := range setters {
+		s(&opts)
+	}
+
+	policies := opts.Policies
+	if policies == nil {
+		policies = MergePolicies{}
+	}
+
 	for key, val := range args {
-		a[key] = val
+		policy := policies[key]
+
+		switch policy {
+		case MergeDenied:
+			return NewDenylistError(key)
+		case MergeAdditive:
+			values := strings.Split(a[key], ",")
+			definedValues := map[string]struct{}{}
+
+			for i, v := range values {
+				definedValues[strings.TrimSpace(v)] = struct{}{}
+
+				if v == "" {
+					values = append(values[:i], values[i+1:]...)
```

Comment:
I feel this is not valid - removing items from the slice during the iteration.

Proper way to filter while iterating is https://github.com/golang/go/wiki/SliceTricks#filter-in-place